client/admin/components/users/ImportDialog.vue

```diff
@@ -0,0 +1,168 @@
+          <v-spacer/>
+          <v-fade-transition>
+            <v-btn
+              v-show="visible && errors"
+              @click="downloadErrorsFile"
+              color="error">
+              <v-icon>cloud_download</v-icon>Errors
+            </v-btn>
+          </v-fade-transition>
+          <v-btn @click="close">Cancel</v-btn>
+          <v-btn :disabled="disabled" color="success" type="submit">
+            Import
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-form>
+  </v-dialog>
+</template>
+
+<script>
+import api from '@/admin/api/user';
+import CircularProgress from '@/common/components/CircularProgress';
+import JSZip from 'jszip';
+import saveAs from 'save-as';
+import { withFocusTrap } from '@/common/focustrap';
+import { withValidation } from '@/common/validation';
+
+const el = vm => vm.$children[0].$refs.dialog;
+const rules = {
+  required: true,
+  mimes:
+  [
+    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    'application/vnd.ms-excel',
+    'text/csv'
+  ]
+};
+
+export default {
+  name: 'import-dialog',
+  mixins: [withValidation(), withFocusTrap({ el })],
+  data() {
+    return {
+      visible: false,
+      importing: false,
+      disabled: true,
+      filename: null,
+      form: null,
+      errors: null,
+      rules: rules
+    };
+  },
+  computed: {
+    showErrors() {
+      return this.errors
+        ? ['There were some erros.']
+        : this.vErrors.collect('file');
+    }
+  },
+  methods: {
+    onFileSelected(e) {
+      this.form = new FormData();
+      this.errors = null;
+      const [file] = Array.from(e.target.files);
+      if (!file) {
+        this.filename = null;
+        return (this.disabled = true);
+      }
+      this.filename = file.name;
+      this.form.append('file', file, file.name);
+      this.$validator.validateAll().then(isValid => {
+        if (!isValid) return (this.disabled = true);
+        this.disabled = false;
+      });
+    },
+    close() {
+      this.disabled = true;
+      this.filename = null;
+      this.$refs.fileInput.value = null;
+      this.visible = false;
+    },
+    save() {
+      this.disabled = true;
+      this.$validator.validateAll().then(isValid => {
+        if (!isValid || this.errors) return;
+        this.importing = true;
+        return api.bulkImport(this.form).then(response => {
+          this.importing = false;
+          if (response.data.size) {
+            this.$nextTick(() => this.$refs.textFile.focus());
+            return (this.errors = response.data);
+          }
+          this.$emit('imported');
+          this.close();
+        }).catch(err => {
+          this.importing = false;
+          this.errors = true;
+          this.$nextTick(() => this.$refs.textFile.focus());
+          return Promise.reject(err);
+        });
+      });
+    },
+    downloadErrorsFile() {
+      JSZip.loadAsync(this.errors)
+        .then(zip => zip.generateAsync({ type: 'blob' }))
```

Comment:
Help me understand this one :thinking: `JSZip` is used here to load zip blob? and return it without modifying it? :confused:

Reply:
Btw couldn't you just return error file as part of server response thus avoiding increasing clientside bundle by `JSZip` inclusion? :thinking: